File: frontend/src/services/adminService.js
import api from './api'

export async function getAdminStats() {
  const response = await api.get('/admin/stats')
  return response.data.data
}

export async function getAllUsers() {
  const response = await api.get('/admin/users')
  return response.data.data
}

export async function updateUserRole(userId, role) {
  const response = await api.patch(`/admin/users/${userId}/role?role=${role}`)
  return response.data.data
}

export async function deleteEvent(eventId) {
  const response = await api.delete(`/admin/events/${eventId}`)
  return response.data
}

export async function deleteLostFoundItem(itemId) {
  const response = await api.delete(`/admin/lost-found/${itemId}`)
  return response.data
}

File: backend/src/main/java/com/campusone/service/AdminService.java
package com.campusone.service;

import com.campusone.dto.response.AdminStatsResponse;
import com.campusone.dto.response.UserResponse;
import com.campusone.entity.Role;

import java.util.List;

public interface AdminService {
    AdminStatsResponse getSystemStats();
    List<UserResponse> getAllUsers();
    UserResponse updateUserRole(Long userId, Role role);
    void deleteEvent(Long eventId);
    void deleteLostFoundItem(Long itemId);
}
